sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2616,6 +2617,21 @@ def check_resources_fit_cluster(
         # was handled by ResourceHandle._update_cluster_region.
         assert launched_resources.region is not None, handle
 
+        # Check whether Azure cluster uses Azure ML API
+        if launched_resources.cloud.is_same_cloud(clouds.Azure()):
+            task_use_az_ml = skypilot_config.get_nested(('azure', 'use_az_ml'),
+                                                        False)
+            cluster_use_az_ml = launched_resources.use_az_ml
+            if cluster_use_az_ml != task_use_az_ml:
+                task_str = 'uses' if task_use_az_ml else 'does not use'
+                cluster_str = 'uses' if cluster_use_az_ml else 'does not use'
+                with ux_utils.print_exception_no_traceback():
+                    raise exceptions.ResourcesMismatchError(
+                        f'Task requirements {task_str} Azure ML API, but the '
+                        f'specified cluster {cluster_name} {cluster_str} it. '
+                        f'Please set azure.use_az_ml to {cluster_use_az_ml} in '
+                        '~/.sky/config.yaml.')
+
```

Comment:
I think we should use the existing cluster's setting even if the config has changed to use ML API. This is to be consistent with other part of the code, e.g., if a user specify GCP DWS, and having a existing cluster not using DWS, we should still allow task to be submitted to that cluster.